File: net/pdfjet/SquareAnnotation.cs
/*
 * SquareAnnotation.cs
 *
 * Copyright (c) 2026 PDFjet Software
 * Licensed under the MIT License. See LICENSE file in the project root.
 */
using System;

namespace PDFjet.NET {
/// <summary>A square annotation.</summary>
public class SquareAnnotation : BaseAnnotation {
    /// <summary>Creates a square annotation.</summary>
    public SquareAnnotation() {
        base.annotationType = Annotation.Square;
    }
}
}

File: src/squareannotation.go
// squareannotation.go
//
// Copyright (c) 2026 PDFjet Software
// Licensed under the MIT License. See LICENSE file in the project root.

package pdfjet

// SquareAnnotation extends BaseAnnotation to represent a square.
type SquareAnnotation struct {
	BaseAnnotation
}

// NewSquareAnnotation acts as the constructor.
func NewSquareAnnotation() *SquareAnnotation {
	s := &SquareAnnotation{BaseAnnotation: *newBaseAnnotation()}
	s.BaseAnnotation.annotationType = annotationSquare
	return s
}
